### PR TITLE
Fix `<ReferenceArrayInput>` should forward `queryOptions` to `getMany`

### PR DESCRIPTION
**Pull request by bendenoz**
fix for #9953

### CODE REVIEW

**Review by slax57 — APPROVED** [review]

**Review by slax57 — CHANGES_REQUESTED** [review]
Thanks for the PR :slightly_smiling_face: 

Unfortunately this seems to throw a TS error at build time. Can you check?

**Comment by bendenoz** [comment]
I rebased on 4.x, is it better?

**Comment by slax57** [comment]
Thanks for the rebase.
Unfortunately the issue remains.
This is due to the `queryOptions` being typed as options for the `getList` query, which is not compatible with the `getManyAggregate` query.

It seems I do not have push permissions on your branch, so I published the required change here: https://github.com/marmelab/react-admin/commit/d7ed98ecb729456a178b12c2b0ff180cb1152ee2
This commit makes it so that the `queryOptions` are typed for `getList` query **or** `getManyAggregate` query.

Can you apply it to your branch?

Thanks.

**Comment by bendenoz** [comment]
cherry-picked and pushed

**Comment by slax57** [comment]
Thanks! :slightly_smiling_face: 

I'm sorry I didn't realize this before, but it would be nice to also update the [documentation](https://github.com/marmelab/react-admin/blob/4.x/docs/ReferenceArrayInput.md#queryoptions) to specify that the `queryOptions` are passed to both `getList` and `getMany` now.

Can you update it? :pray:

**Comment by bendenoz** [comment]
ok :) see latest commit, hopefully it' only in one place?

File: docs/ReferenceArrayInput.md [hunk 1]
```diff
@@ -254,7 +254,7 @@ This prop is mostly useful when using [`<SelectArrayInput>`](./SelectArrayInput.
 
 ## `queryOptions`
 
-Use the `queryOptions` prop to pass options to the `dataProvider.getList()` query that fetches the possible choices.
+Use the `queryOptions` prop to pass options to the `dataProvider.getList()` query that fetches the possible choices, as well as the `dataProvider.getManyAggregate()` query that retrieves the currently selected choices.
```
Comment (slax57) [inline docs/ReferenceArrayInput.md:257]:
```suggestion
Use the `queryOptions` prop to pass options to the `dataProvider.getList()` query that fetches the possible choices, as well as the `dataProvider.getMany()` query that retrieves the currently selected choices.
```

**Review by slax57 — COMMENTED** [review]
Thanks! I think we're good :slightly_smiling_face: (apart from this last suggestion)

**Review by slax57 — APPROVED** [review]
Thanks!

**Comment by lumpov** [comment]
please release this fix in v4

**Comment by fzaninotto** [comment]
@lumpov Sorry, this is a new feature and v4 is feature freeze. You'll have to upgrade to v5 if you need this feature.

**Pull request merged**